Units/regex-with-scope.d/expected.tags

```diff
@@ -0,0 +1,20 @@
+NS1	input.foo	/^ns NS1$/;"	n
+x	input.foo	/^define x {$/;"	d	namespace:NS1
+y	input.foo	/^	define y {$/;"	d	definition:x
+v_y_0	input.foo	/^	       var v_y_0$/;"	v	definition:y
```

Comment:
Tank you. You are right. Very interesting.

```
[jet@localhost]~/var/ctags% cat /tmp/foo.py 
cat /tmp/foo.py 
def x():
    def y():
        def z():
            pass
[jet@localhost]~/var/ctags% ./ctags -o - /tmp/foo.py 
./ctags -o - /tmp/foo.py 
x	/tmp/foo.py	/^def x():$/;"	f
y	/tmp/foo.py	/^    def y():$/;"	f	function:x	file:
z	/tmp/foo.py	/^        def z():$/;"	f	function:x.y	file:
[jet@localhost]~/var/ctags% cat /tmp/foo.c
cat /tmp/foo.c
struct x {
  struct y {
    int a;
  } z;
};
[jet@localhost]~/var/ctags% ./ctags -o - /tmp/foo.c
./ctags -o - /tmp/foo.c
a	/tmp/foo.c	/^    int a;$/;"	m	struct:x::y	file:
x	/tmp/foo.c	/^struct x {$/;"	s	file:
y	/tmp/foo.c	/^  struct y {$/;"	s	struct:x	file:
z	/tmp/foo.c	/^  } z;$/;"	m	struct:x	typeref:struct:x::y	file:
```

It seems that full qualified tag is needed as scope name.
Scope context is tracked with a stack, it may be possible to make full qualified tag.
The basic idea is already show in #524.I have some code.

One of ths issue is the command line interface to specify scope separator(operator or combinator?).
You wrote `definitions:NS1.x.y`. However, if `.` is for string concatenation in this language,
the user may want ... `definitions:NS::x->y` or something natural in the language. The worst
case is that a language allows `.` being part of a name.

In addition, a language neutral tool may want to know which characters (or strings) are the separator for the tag to separate the each components of a scope name.

How about following one:
```
--separator-<LANG>=+<KIND1><KIND2><SEPARATOR>
```
Applying this to foo:
```
--separator-foo=+nd::
--separator-foo=+dd->
```
We need a wild card. Kind letter `*` must be ctags side.

About the output, ... we may have to embed it in pseudo tags.
```
!_TAG_LANGUAGE_SEPARATOR\tnd\t/::/;"language:foo
!_TAG_LANGUAGE_SEPARATOR\tdd\t/->/;"language:foo
```

I have never thought about pseudo tags well so more considerations are needed.
As the initial step I will introduce `.` as default value for scope separator.